utilities/tools/platform/Set-ApiSpecsFile.ps1

```diff
@@ -0,0 +1,49 @@
+﻿<#
+.SYNOPSIS
+Update the API Specs file in the given path
+
+.DESCRIPTION
+Update the API Specs file in the given path. The file contains an outline of all Provider Namespaces with their Resource Types and supported API versions.
+
+.PARAMETER SpecsFilePath
+Optional. The path the the file to create/overwrite. By default points to path '<root>/utilities/src/apiSpecsList.json'
+
+.EXAMPLE
+Set-ApiSpecsFile -SpecsFilePath 'C:/dev/ResourceModules/utilities/src/apiSpecsList.json'
+
+Update the file in path 'C:/dev/ResourceModules/utilities/src/apiSpecsList.json' with the latest API versions.
+#>
+function Set-ApiSpecsFile {
+
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    param (
+        [Parameter(Mandatory = $false)]
+        [string] $SpecsFilePath = (Join-Path (Split-Path (Split-Path $PSScriptRoot -Parent)) 'src' 'apiSpecsList.json')
+    )
+
+    # Install and or import module
+    if (-not (Get-Module 'AzureAPICrawler' -ListAvailable)) {
+        if ($PSCmdlet.ShouldProcess("Module 'AzureAPICrawler with version [0.2.0]'", 'Install')) {
+            $null = Install-Module 'AzureAPICrawler' -Scope 'CurrentUser' -Repository 'PSGallery' -RequiredVersion '0.2.0' -Force
+        }
+    }
+
+    $null = Import-Module 'AzureAPICrawler'
+
+    # Fetch data
+    $res = Get-AzureApiSpecsVersionList -IncludePreview -Verbose
```

Comment:
Same for includePreview, this should probably be an input parameter. We could even discuss if those values should be passed through the repo settings in a next PR.

Reply:
I was just about to update it - however, half way through I was wondering if this should be passed or - or instead much rather be something to configure for the test itself? - The 'Data Source' can always be complete -  but the test could then restrict to only allow non-preview for example.

Reply:
Justed updated it this way and will test it

Reply:
tested and works

Reply:
> allow non-preview for example.

Is that something we provide as an option to users internalizing the CI environment? If so I think we lack wiki guidelines on how to allow non-preview only if needed. 

Tricky topic anyway. One may easily have the wrong impression that preview API version = preview service

Reply:
We provide a flag in the settings alongside documentation what that means

Reply:
My bad, I didn't notice the settings.yml update.

Reply:
You shall be forgiven